migration-preset/index.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+const preset = require('neutrino-preset-react');
+const HtmlPlugin = require('html-webpack-plugin');
+const webpack = require('webpack');
+const path = require('path');
+const fs = require('fs');
+const yaml = require('js-yaml');
+const _ = require('lodash');
+
+const WORKGRAPH_DIR = 'workgraph';
+
+class WorkgraphDataPlugin {
+  apply(compiler) {
+    compiler.plugin('emit', (compilation, callback) => {
+      const graph = this.readGraph();
+
+      // record dependencies on file contents
+      _.forEach(graph.inputs, (filename) => {
+        compilation.fileDependencies.push(filename)
+      });
+
+      // include an asset containing the graph JSON
+      const graphJson = "WORKGRAPH = " + JSON.stringify(graph.data);
+      compilation.assets['workgraph.js'] = {
+        source: () => graphJson,
+        size: () => graphJson.length
+      };
+      callback();
+    });
+  }
+
+  readGraph() {
+    const inputs = [];
+
+    _.forEach(fs.readdirSync(WORKGRAPH_DIR), (filename) => {
```

Comment:
I added a `console.log` and only saw emit run once per compilation.  I can't find any information about what emit does (other than that the compilation is "sealed" by that time, which seems weird since I can still add assets..), so I don't know -- is it called once per output file or something?